### DISCUSSION TITLE
502 Error with configuration

### DISCUSSION THREAD

**Discussion by RichardOelschlager** [pretix/pretix#2849] [Q&A]
server {

        root /var/www/domain.com/html;
        index index.html index.htm index.nginx-debian.html;

        server_name domain.com www.domain.com pretix.domain.com;

         location / {
        proxy_pass http://localhost:8345;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Host $http_host;
    }

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/domain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = pretix.domain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot
        if ($host = domain.com) {
                return 301 https://$host$request_uri;
}

        listen 80;
        listen [::]:80;

        server_name domain.com www.domain.com pretix.domain.com;
    return 404; # managed by Certbot

}

**Comment by raphaelm** [comment]
Please check the nginx error log for a reason for the 502 error

**Comment by RichardOelschlager** [comment]
2022/10/12 06:46:16 [error] 11045#11045: *678 upstream timed out (110: Connection timed out) while reading response hea>
2022/10/12 06:46:17 [crit] 11045#11045: *700 SSL_do_handshake() failed (SSL: error:141CF06C:SSL routines:tls_parse_ctos>
2022/10/12 06:46:26 [error] 11045#11045: *680 upstream timed out (110: Connection timed out) while reading response hea>
2022/10/12 06:46:44 [error] 11045#11045: *684 upstream timed out (110: Connection timed out) while reading response hea>
2022/10/12 06:46:53 [error] 11045#11045: *686 upstream timed out (110: Connection timed out) while reading response hea>

**Reply by raphaelm** [reply]
the problem is probably with the pretix process then, either it's not running at localhost:8345 or it's not responding. any interesting log messages there?

**Reply by RichardOelschlager** [reply]
And where is this log file located? Sorry, very much a beginner.

**Reply by raphaelm** [reply]
see https://docs.pretix.eu/en/latest/admin/errors.html for some pointers :)

**Reply by RichardOelschlager** [reply]
WARNING 2022-10-11 22:38:26,616 django.request log Bad Request: /nice ports,/Trinity.txt.bak
WARNING 2022-10-11 22:38:26,616 django.request log Bad Request: /nice ports,/Trinity.txt.bak
WARNING 2022-10-11 22:38:26,971 django.request log Bad Request: /
WARNING 2022-10-11 22:38:26,971 django.request log Bad Request: /
WARNING 2022-10-12 00:48:50,890 django.request log Bad Request: /
WARNING 2022-10-12 00:48:50,890 django.request log Bad Request: /
WARNING 2022-10-12 01:55:51,594 django.request log Bad Request: /favicon.ico
WARNING 2022-10-12 01:55:51,594 django.request log Bad Request: /favicon.ico
WARNING 2022-10-12 02:41:48,879 django.request log Bad Request: /
WARNING 2022-10-12 02:41:48,879 django.request log Bad Request: /
WARNING 2022-10-12 11:56:14,730 django.request log Bad Request: /favicon.ico
WARNING 2022-10-12 11:56:14,730 django.request log Bad Request: /favicon.ico
